### PR TITLE
Use Authorization headers for public webdav in web UI

### LINKED ISSUE #21196: Public link page not working in IE9 with Webdav
### Steps

1. Setup IE9 in a VM
1. Set trusted domain to "192.168.56.1"
1. Login in the VM
1. Create a folder "test"
1. Share "test" with link
1. Open link (still in the VM)

### Expected result
File list shown

### Actual result
Endless spinner and "security error":
ie9-sec-problem

I tried logging the URL, but the URL for the XMLHttpRequest is in the same domain so not sure what IE9 is complaining about... The regular file list works fine.

### Versions
ownCloud 9.0 pre-alpha git master 74de12c698e01514c8fe42727d98aeaa8956eeee

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/21196

Now using Authorization headers instead of prepending the token in the URL. This, because IE9 and co would believe it's a cross-domain call and refuse to do it...

Need to public link with password in:
- [x] IE8
- [x] IE9
- [x] IE10
- [x] IE11
- [x] Safari
- [x] Firefox
- [x] Chromium

### CODE REVIEW

**Comment by PVince81** [comment]
CC @DeepDiver1975 for the addition of a new polyfill library specifically for btoa/atob (only loaded for IE and friends)

**Comment by PVince81** [comment]
@LukasReschke no objection for using Authorization headers from the browser ?

File: core/vendor/base64/LICENSE [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+            DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
+                    Version 2, December 2004
```
Comment (PVince81) [inline core/vendor/base64/LICENSE]: talked with @schiesbn, license is compatible :smile:

**Comment by MorrisJobke** [comment]
I tested this with password protected link shares to folders and to files in native IE8, IE9, IE11, Edge, Firefox, Chrome and Safari. IE10 was tested with the build in "use IE10 rendering engine" in Windows 10 Internet Explorer.

I also tested that link shares to folders and file still work without problems.

All works :+1:

**Comment by PVince81** [comment]
Needs a second reviewer @icewind1991 @LukasReschke @DeepDiver1975

**Comment by icewind1991** [comment]
:+1: looks good, tested in chrome and edge

**Pull request merged**